subsys/bluetooth/mesh/light_ctrl_srv.c

```diff
@@ -421,7 +421,7 @@ static void transition_start(struct bt_mesh_light_ctrl_srv *srv,
 	}
 	restart_timer(srv, fade_time);
 #if CONFIG_BT_MESH_LIGHT_CTRL_SRV_REG
-	if (srv->reg && atomic_test_bit(&srv->flags, FLAG_AMBIENT_LUXLEVEL_SET)) {
+	if (srv->reg && is_enabled(srv)) {
```

Comment:
`is_enabled` is not needed here because `transition_start` can't be called when controller is not enabled. Previous check was enough I think since it is now set only when regulator is enabled.

Reply:
With new logic and old check, the 'target' value can be missed, if the regulator happens to be turned on just after setting Mode=1. For example, Send Light LC Mode Set (mode=1), and quickly send Light LC Light ON (TT=0), now if sensor message takes a second or two to arrive (sensor reacts slowly), then the old condition will evaluate to false. This will make "target" of the regulator to not get set to the appropriate value. So, when sensor messages does arrive, now regulator runs with the stale target.

Reply:
Right, but then still not sure why `is_enabled` is needed.

Reply:
That's true. Just to avoid additional assignments? Removing `is_enabled()` is actually better.

Reply:
Yes, I think `if (srv->reg)` is enough/